### ISSUE TITLE
$user in search path causes exception

### ISSUE THREAD

**Issue by dpmillerau** [denishpatel/pg-clone-schema#47]
$user in the search path causes an exception right at the end of the function!

**Comment by dpmillerau** [comment]
solution - quote the string 
https://github.com/dpmillerau/pg-clone-schema/commit/04c7828b30c6ac54c8e4e3c5d759a4ec7cb36a09

**Comment by MichaelDBA** [comment]
Thanks again for the fix suggestion.  It is now implemented.

**Comment by guignonv** [comment]
This fix does not work as expected. The search path is not set back to it's original value but a full quoted string containing the original value. `quote_literal()` will return a string while `SET search_path = ` expects a list of schema identifiers.

Let's say I got this:
```
=> SELECT setting FROM pg_settings WHERE name='search_path';
        setting
------------------------
 "$user", public, someotherschema
(1 row)
```
and it is stored into 'src_path_old', then this:
```=> EXECUTE 'SET search_path = ' || quote_literal(src_path_old);```
will lead to 
```
=> SELECT setting FROM pg_settings WHERE name='search_path';
        setting
------------------------
"""$user"", public, someotherschema"
(1 row)
```
The whole thing is quoted and then, no relation is found by PostgreSQL since the new search_path does not exist. It will not look into `$user`, then `public` and finally `someotherschema` but rather in `""$user"", public, someotherschema` schema which does not exists.

From my investigations, the problem is not trivial as `clone_schema()` works well with `EXECUTE 'SET search_path = ' || src_path_old;` when I run it form PostgreSQL interactive console, but when I call it from a PHP application, I got the exception as described in this issue. If I use `EXECUTE 'SET search_path = ' || quote_literal(src_path_old);`, it will appear to work (no exception) but the search path is not right and then, any other query using an unqualified database object will fail (both interactive console and PHP script). I guess, in some way, the double quotes around `$user` are removed by PHP or the PostgreSQL drivers. In order to fix that for both interfaces (PostgreSQL interactive console and applications), I just added an expression that guaranties to have $user double quoted just once, when we get the original search_path:
`SELECT REPLACE(REPLACE(setting, '"$user"', '$user'), '$user', '"$user"') INTO src_path_old FROM pg_settings WHERE name = 'search_path';`
It worked for me from both the console and PHP. I'll submit a PR.
